src/firmware/main/math/polynomial_1d.c

```diff
@@ -0,0 +1,67 @@
+#include "firmware/main/math/polynomial_1d.h"
+
+#include <math.h>
+
+/**
+ * Compute the value at the given point for the polynomial formed by the given coeffs
+ * @param coefficients The coefficients that makeup the polynomial, with the highest
+ *                     order coefficients at the start
+ * @param num_coefficients The number of coefficients
+ * @param x The point to get the polynomial value at
+ * @return The polynomial value at the given point
+ */
+float shared_polynomial1d_getValue(const float* coefficients,
+                                   const size_t num_coefficients, float x)
+{
+    float result = 0;
+    for (size_t order = 0; order < num_coefficients; order++)
+    {
+        result += coefficients[num_coefficients - 1 - order] * pow(x, order);
+    }
+    return result;
+}
+
+/**
+ * Calculate the derivative of the polynomial given by the given coefficients
+ *
+ * @param coefficients [in] The coefficients representing the polynomial to differentiate,
+ *                          with the highest order ones first.
+ * @param num_coefficients [in] The number of coefficients
+ * @param derivative_coefficients [out] This will be populated with the coefficients
+ *      representing the derivative of the polynomial represented by the given
+ *      coefficients. Must be allocated to be at least num_coefficients-1 long
+ */
+void shared_polynomial1d_differentiate(const float* coefficients, size_t num_coefficients,
```

Comment:
Just noting that const is only protecting the pointer iinm

Reply:
Yep, I generally try to make everything const if possible as a habit, just because it makes it easier to reason about the code if you can guarantee variables won't change, but you're totally correct.